assets/js/base/components/button/editor.scss

```diff
@@ -0,0 +1,14 @@
+// Most of these styles are the same as in `style.scss`,
+// but we need to give them more specificity in the editor.
+.edit-post-visual-editor .button.wc-block-button {
+	background-color: #000 !important;
+	color: #fff !important;
+	height: auto;
```

Comment:
I realize this isn't new, but I just realized that this is for the `base/components/button` component. It seems like this should be a button specific to cart and checkout blocks if we're using `!important` like this because it means we're explicitly locking down the background and foreground colors for this button.

I would _expect_ something like `base/components/button` to be a bit more generic.  On the other hand, is this _intentional_? Are we intentionally declaring here that any component implementing this button (`wc-block-button` component) is getting this behaviour?

The comments above don't block this pull, but I do wonder if we should consider moving this to `base/components/cart-checkout/button/*`?

Reply:
Another potential problem with using `!important` here, is I see `style.css` doesn't have this flag used. That means if a theme styles the button differently it won't be reflected in the editor.

I think we _should_ lock down the styling for the button in both contexts to match the wireframes (meaning themes can't change the button colors) and in later iterations we could explore making this configurable via block options (maybe the button would become a block). So I think it might be prudent (followup) to move this component to `cart-checkout/button` (as mentioned previously) and force styling for both contexts. Thoughts?

Reply:
Thanks for the review @nerrad. Good points with everything.

> The comments above don't block this pull, but I do wonder if we should consider moving this to `base/components/cart-checkout/button/*`?

Just to make sure I understand you, you mean creating a subfolder under `base/components` for `cart-checkout` components? In that case, would make it sense moving `address-form`, `shipping-calculator`, `shipping-location`, `shipping-rates-control`, `sidebar-layout` and `totals` too? There are some others which are only used in the _Cart_ & _Checkout_ block but are more generic like `country-input`, `tabs`, etc. that I'm not sure if they should be in the `cart-checkout` subfolder or in the `base/components` root.

Not for now but maybe for the cooldown, would make it sense grouping the components used by the _All Products_ & Filters blocks and _Reviews_ blocks into subfolders as well? I see `base/components` is growing quite fast and might be good to add some hirerarchy there.

Reply:
> In that case, would make it sense moving address-form, shipping-calculator, shipping-location, shipping-rates-control, sidebar-layout and totals too?

Yes, exactly. Anything that's _specific_ to a top-level component and not for re-use would go in the sub-folder. 

> Not for now but maybe for the cooldown, would make it sense grouping the components used by the All Products & Filters blocks and Reviews blocks into subfolders as well? 

Ya that'd be a good idea. But we'd have to be careful that we're only putting components in there that wouldn't be re-used in any other blocks.

Reply:
Ok, in ac390623f7c49204c9657c1b28fa54f940da8ad2 I merged `editor.scss` and `style.scss` files so we have a single source of truth for styles. In order to get rid of the `!important` rules we would need to stop using `<Button>` from `@wordpress/components`. I think that's something that we should consider at some point, our implementation of `<Button>` styles differs too much from what's done in `@wordpress/components` and IMO having CSS resets makes our codebase more complex and difficult to maintain than just creating a `<button>` component from scratch.

In 181fac3a53603ec2b3dcf8454cc2c7459e5dffa1, I moved `<Button>` inside the `cart-checkout` folder as suggested.

I also created an issue (#1962) to track moving more components into its subfolders. I assigned it to myself and added the cooldown label, but feel free to remove it if you think that's something that's better suited for a regular cycle. :+1:

Reply:
All looks good! Thanks Albert, let's :shipit: 🚀